cartesian_pose_impedance_controller:
    type: franka_interactive_controllers/CartesianPoseImpedanceController
    arm_id: panda
    joint_names:
        - panda_joint1
        - panda_joint2
        - panda_joint3
        - panda_joint4
        - panda_joint5
        - panda_joint6
        - panda_joint7

cartesian_twist_impedance_controller:
    type: franka_interactive_controllers/CartesianTwistImpedanceController
    arm_id: panda
    joint_names:
        - panda_joint1
        - panda_joint2
        - panda_joint3
        - panda_joint4
        - panda_joint5
        - panda_joint6
        - panda_joint7

passive_ds_impedance_controller:
    type: franka_interactive_controllers/PassiveDSImpedanceController
    arm_id: panda
    joint_names:
        - panda_joint1
        - panda_joint2
        - panda_joint3
        - panda_joint4
        - panda_joint5
        - panda_joint6
        - panda_joint7


joint_gravity_compensation_controller:
    type: franka_interactive_controllers/JointGravityCompensationController
    arm_id: panda
    joint_names:
        - panda_joint1
        - panda_joint2
        - panda_joint3
        - panda_joint4
        - panda_joint5
        - panda_joint6
        - panda_joint7


joint_position_franka_controller:
    type: franka_interactive_controllers/JointPositionFrankaController
    joint_names:
        - panda_joint1
        - panda_joint2
        - panda_joint3
        - panda_joint4
        - panda_joint5
        - panda_joint6
        - panda_joint7

joint_velocity_franka_controller:
    type: franka_interactive_controllers/JointVelocityFrankaController
    joint_names:
        - panda_joint1
        - panda_joint2
        - panda_joint3
        - panda_joint4
        - panda_joint5
        - panda_joint6
        - panda_joint7

joint_impedance_franka_controller:
    type: franka_interactive_controllers/JointImpedanceFrankaController
    arm_id: panda
    joint_names:
        - panda_joint1
        - panda_joint2
        - panda_joint3
        - panda_joint4
        - panda_joint5
        - panda_joint6
        - panda_joint7
    k_gains:
        - 600.0
        - 600.0
        - 600.0
        - 600.0
        - 250.0
        - 150.0
        - 50.0
    d_gains:
        - 50.0
        - 50.0
        - 50.0
        - 20.0
        - 20.0
        - 20.0
        - 10.0
    radius: 0.1
    acceleration_time: 2.0
    vel_max: 0.15
    publish_rate: 10.0
    coriolis_factor: 1.0

cartesian_pose_franka_controller:
    type: franka_interactive_controllers/CartesianPoseFrankaController
    arm_id: panda

cartesian_velocity_franka_controller:
    type: franka_interactive_controllers/CartesianVelocityFrankaController
    arm_id: panda
        
cartesian_force_controller:
    type: franka_interactive_controllers/CartesianForceController
    arm_id: panda
    joint_names:
        - panda_joint1
        - panda_joint2
        - panda_joint3
        - panda_joint4
        - panda_joint5
        - panda_joint6
        - panda_joint7
